Code/Tools/AssetProcessor/native/tests/assetmanager/AssetProcessorManagerTest.cpp

```diff
@@ -4130,11 +4131,20 @@ struct LockedFileTest
     MOCK_METHOD2(SendResponse, size_t (unsigned, const AzFramework::AssetSystem::BaseAssetProcessorMessage&));
     MOCK_METHOD1(RemoveResponseHandler, void (unsigned));
 
-    size_t Send(unsigned, const AzFramework::AssetSystem::BaseAssetProcessorMessage&) override
+    size_t Send(unsigned, const AzFramework::AssetSystem::BaseAssetProcessorMessage& message) override
     {
-        if(m_callback)
+        using SourceFileNotificationMessage = AzToolsFramework::AssetSystem::SourceFileNotificationMessage;
+        switch (message.GetMessageType())
         {
-            m_callback();
+        case SourceFileNotificationMessage::MessageType:
+            if (const auto sourceFileMessage = static_cast<const SourceFileNotificationMessage&>(message);
```

Comment:
Shouldn't this be an `azrtti_cast`?

Reply:
Yes it should be.

Reply:
Actually there is an issue with `azrtti_cast`.
It can't be used to cast a reference.